### ISSUE TITLE
Allow automatically generating TLS certificates

### ISSUE THREAD

**Issue by Nothing4You** [MariaDB/mariadb-docker#405]
The mysql image automatically generates TLS certificates on startup if there is no key present.
It would be useful to have this functionality in the mariadb image as well, especially for testing in CI runs.

For example using GitHub Actions [service containers](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) does not provide an easy way to bring your own pre-generated certificates from your repository.

https://github.com/docker-library/mysql/blob/37cf4047a0544046c9ecce8f2885d4f362a227bc/template/docker-entrypoint.sh#L200-L205

As there does not seem to be an equivalent `mariadb_ssl_rsa_setup` this would probably need certificates generated explicitly via e.g. `openssl`.

**Comment by grooverdan** [comment]
Good idea. From [MDEV-16219](https://jira.mariadb.org/browse/MDEV-16219) its obviously an outstanding feature request that has been there a while. I can't say when I get to it but making the feature available for test and production systems sounds like a really good feature.

For your https://github.com/Nothing4You/aiomysql/pull/1, you may be interested in https://mariadb.org/new-service-quay-io-mariadb-foundation-mariadb-devel/

**Comment by grooverdan** [comment]
Note: https://mariadb.org/mission-impossible-zero-configuration-ssl/ 11.4+
